### DISCUSSION TITLE
unable to update from 12.2.0 to 13.3.0

### DISCUSSION THREAD

**Discussion by i-am-rumburak** [arendst/Tasmota#20386] [General]
i've started playing with 3 gosund sp111 v1.1 sockets that I bought some time ago with preloaded tasmota. I went into webgui -> information and it shows 12.2.0. noticed firmware upgrade, clicked ota update, it does something, restarts but eventually still shows 12.2.0. what am I missing? I've read somewhere in docs that upgrading via OTA is simplest way to do but it looks like it does nothing, doesn't upgrade.

**Comment by sfromis** [comment]
Without details, a guess: Read https://github.com/arendst/Tasmota/discussions/18983

**Reply by barbudor** [reply]
@sfromis No, SP111 are ESP8266 based

@i-am-rumburak : Are you sure your SP111 has access to the internet ? Proper gateway and proper DNS ?

**Reply by i-am-rumburak** [reply]
@sfromis what details would you like to see? I will do my best to provide them.

@barbudor hmm. they are connected to my wi-fi, I see that network is set properly, they are assigned DHCP addresses, tasmota-remota sees them. if there is a way to be sure 100% (like ping outside or something) please let me know. at least there was no error like "network failure" or "cannot connect to server". I also downloaded tasmota package manually, selected it below OTA update, again it did something, restarted but nothing changed either.

please let me know what you'd like to see and I'll do my best to get it.

**Reply by barbudor** [reply]
try the following command
```
WebSend [ota.tasmota.com] /
```
The response should be 
```
{"WebSend":"Done"}
```
If you get "Connect failed", either you have a wrong DNS or a wrong gateway

In that case try
```
websend [1.1.1.1] /
```
If you still have "Connect failed" then the problem is your gateway

**Reply by i-am-rumburak** [reply]
I see `{"WebSend":"Done"}`

[https://ibb.co/FXQJky6](https://ibb.co/FXQJky6)

**Reply by barbudor** [reply]
ok so you have proper access to the Internet

What is the URL in your "OTA Url" ?
You should have either
- `http://ota.tasmota.com/tasmota/tasmota.bin.gz` for the latest development version or
- `http://ota.tasmota.com/tasmota/release/tasmota.bin.gz` for the latest release

Be sure the URL ends with `.bin.gz`

One thing you can try is open 2 web browser windows side by side (or one on a computer, and one on a phone)
- One on the upgrade page
- One on the web console
Click on the "Start upgrade" button and check the logs on the other window

![image](https://github.com/arendst/Tasmota/assets/2996042/09008fb7-151d-446e-881f-41d3f436cae8)

**Reply by i-am-rumburak** [reply]
I have the second one:

`http://ota.tasmota.com/tasmota/release/tasmota.bin.gz`

this is also the file I downloaded locally and pointed in "upgrade by file upload"

will try do do logging and let you know

**Reply by barbudor** [reply]
If you want to do it locally, not through "Upgrade by webserver" by through "Upgrade by file upload" you will need 2 files:
1. http://ota.tasmota.com/tasmota/release/tasmota-minimal.bin.gz
2. http://ota.tasmota.com/tasmota/release/tasmota.bin.gz

You need to 1st upgrade to "tasmota-minimal"
And then after reboot to the full tasmota

Going through "Upgrade by weberver", those 2 steps occurs in the background without you to intervene

**Reply by i-am-rumburak** [reply]
this is the log:

```
23:00:05.966 RSL: RESULT = {"OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz"}
23:00:06.006 RSL: RESULT = {"Upgrade":"Version 12.2.0 from http://ota.tasmota.com/tasmota/release/tasmota.bin.gz"}
23:00:08.368 HTP: Web server stopped
23:00:09.148 HTP: Web server active on tasmota-5CC3FC-1020 with IP address 192.168.13.52
23:00:10.398 HTP: Web server stopped
23:00:48.667 HTP: Web server active on tasmota-5CC3FC-1020 with IP address 192.168.13.52
23:00:49.868 HTP: Web server stopped
23:01:22.294 HTP: Web server active on tasmota-5CC3FC-1020 with IP address 192.168.13.52
23:01:23.543 HTP: Web server stopped
23:01:53.195 HTP: Web server active on tasmota-5CC3FC-1020 with IP address 192.168.13.52
23:01:54.444 HTP: Web server stopped
23:02:33.959 HTP: Web server active on tasmota-5CC3FC-1020 with IP address 192.168.13.52
23:02:35.209 HTP: Web server stopped
23:03:02.060 HTP: Web server active on tasmota-5CC3FC-1020 with IP address 192.168.13.52
23:03:03.309 HTP: Web server stopped
23:03:04.060 HTP: Web server active on tasmota-5CC3FC-1020 with IP address 192.168.13.52
23:03:05.324 RSL: UPGRADE = {"Upgrade":"Failed Update error: ERROR[6]: Stream Read Timeout"}
```

**Reply by barbudor** [reply]
Hum, strange has it looks like Tasmota cannot download the file

**Reply by i-am-rumburak** [reply]
could it be interferring ad blocker or something? I can download these files in browser.

**Comment by i-am-rumburak** [comment]
**log after minimal:**

```
00:00:00.001 HDW: ESP8285
00:00:00.050 CFG: Loaded from flash at FB, Count 31
00:00:00.055 QPC: Count 1
00:00:00.064 NRG: Init driver 1
00:00:00.067 Project tasmota - Tasmota Version 12.2.0(tasmota)-2_7_4_9(2022-10-17T08:39:23)
00:00:00.557 WIF: Connecting to AP1 XXXXXX Channel 6 BSSId XX:XX:XX:XX:XX:XX in mode 11n as tasmota-5CC3FC-1020...
00:00:01.754 WIF: Connected
00:00:02.007 HTP: Web server active on tasmota-5CC3FC-1020 with IP address 192.168.13.52
23:07:41.017 RSL: INFO1 = {"Info1":{"Module":"Gosund S111 v.1.1","Version":"12.2.0(tasmota)","FallbackTopic":"cmnd/DVES_5CC3FC_fb/","GroupTopic":"cmnd/tasmotas/"}}
23:07:41.023 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-5CC3FC-1020","IPAddress":"192.168.13.52"}}
23:07:41.042 RSL: INFO3 = {"Info3":{"RestartReason":{"Exception":28,"Reason":"Blocked Loop","EPC":["40201b5e","00000000","00000000"],"EXCVADDR":"00000000","DEPC":"00000000","CallChain":["40201b5c","40106109","402512f2","40276848","402520e2","40276855","4027689a","4024e398","40000f49","40000f49","40000e19","40105aa9","40105aaf","4010000d","4026f2cc","4026f27d","40100230","401013fd","401013ec","401013ec","40100230","401013fd","40101324","40100230","401013fd","401013ec","40100230","401013fd","4010342f","40101324","4000050c"]},"BootCount":9}}
23:07:41.081 RSL: RESULT = {"POWER":"ON"}
23:07:41.084 RSL: POWER = ON
23:07:44.448 QPC: Reset
```

**log after 2nd file:**

```
23:09:05.256 UPL: File tasmota.bin.gz
23:09:37.427 UPP: Multicast (re)joined
```

**information**
```

Program Version | 12.2.0(tasmota)
Build Date & Time | 2022-10-17T08:39:23
Core/SDK Version | 2_7_4_9/2.2.2-dev(38a443e)
Uptime | 0T00:03:36
Flash write Count | 33 at 0xFB000
Boot Count | 9
Restart Reason | Blocked Loop

[...]


ESP Chip Id | 6079484 (ESP8285)
Flash Chip Id | 0x144051 (DOUT)
Flash Size | 1024 KB
Program Flash Size | 1024 KB
Program Size | 626 KB
Free Program Space | 376 KB
Free Memory | 24.7 KB


```

**Reply by barbudor** [reply]
You can't even upgrade to minimal
It crashed during the upgrade:
```
INFO3 = {"Info3":{"RestartReason":{"Exception":28,"Reason":"Blocked Loop",
```

I've never seen that on any of my SP111
I fear you have a problem with the flash on your device
You may have to try serial flashing
The SP111 are pretty easy to open by removing the screw from the earth hole
You need to be carefull when removing the PCB not to damage the antenna
pinout for soldering the serial are available here: https://templates.blakadder.com/gosund_SP111_v1_1.html

**Reply by i-am-rumburak** [reply]
i'll try to update other ones (have 3 of them total). a priori seems unlikely all would have flash issue but eventually possible. I'll check the log and let you know.

**Reply by i-am-rumburak** [reply]
the other one has the same log for OTA update:
`23:30:29.674 RSL: UPGRADE = {"Upgrade":"Failed Update error: ERROR[6]: Stream Read Timeout"}`

**Reply by sfromis** [reply]
I'd try doing the upgrade via file upload, instead of the regular way via a http URL.

**Reply by barbudor** [reply]
Already suggested and tested => `INFO3 = {"Info3":{"RestartReason":{"Exception":28,"Reason":"Blocked Loop"`

**Reply by Amdi-de** [reply]
> the other one has the same log for OTA update: `23:30:29.674 RSL: UPGRADE = {"Upgrade":"Failed Update error: ERROR[6]: Stream Read Timeout"}`

Did you resolve it? I‘ve same with a Gosund SP 1 in 12.2.0

21:52:12.799 RSL: UPGRADE = {"Upgrade":"Failed Update error: ERROR[6]: Stream Read Timeout"}
21:53:56.300 RSL: RESULT = {"OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz"}
21:53:56.360 RSL: RESULT = {"Upgrade":"Version 12.2.0 from http://ota.tasmota.com/tasmota/release/tasmota.bin.gz"}

**Comment by i-am-rumburak** [comment]
...and same error as the other one:

`INFO3 = {"Info3":{"RestartReason":{"Exception":28,"Reason":"Blocked Loop"`
